text/0000-advance-security-disclosure.md

```diff
@@ -0,0 +1,305 @@
+
+The security policy will be extended to include two kinds of early
+notifications sent before public disclosure of security vulnerabilities.
+
+The Rust Security Response WG will aim to send them 5 business days ahead of
+public disclosure, or 1 business day ahead of public disclosure in case of
+critical, actively exploited vulnerabilities.
+
+## Advance disclosures
+
+To ensure all Rust users have access to a fixed toolchain, the Rust Security
+Response WG will send the full vulnerability details ahead of time to
+distributors of the Rust toolchain and maintainers of alternate Rust
+implementation (when applicable), giving them all the information required to
+prepare a toolchain update for their users or customers.
+
+Organizations (either companies or established open source projects) eligible
+to be notified will be able to apply for inclusion in the advance disclosure
+list, as long as they meet the requirements outlines in the reference-level
+section of this RFC. Organizations in the list are expected to keep all
+information about those vulnerabilities confidential before public disclosure.
+
+## Security releases pre-announcements
+
+To ensure all Rust users are aware of upcoming security releases, before public
+disclosure the Security Response WG will publish a pre-announcement of the
+upcoming release on the [public security announces mailing list][announce-ml]
+and the [Rust blog][blog], mentioning the planned date and time of the release,
+the severity of the vulnerability and the affected part of the project.
+
+This will allow all Rust users to schedule the time to update their toolchain.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Advance disclosures
+
+### List of notified organizations
+
+The Rust Security Response WG will maintain a list of organizations who are
+eligible to receive advance disclosures. Organizations (either companies or
+established open source projects) matching any of these criteria can apply:
+
+* Organizations distributing the Rust toolchain (as provided by the Rust
+  project) or a fork of the Rust toolchain explicitly meant to be used by their
+  external customers or users. Organizations shipping the toolchain just to
+  internal customers/users are not eligible, nor are organizations publicly
+  shipping a toolchain primarily meant for internal use.
+
+* Organizations developing an independent implementation of the Rust language,
+  not originating from a fork of the toolchain provided by the Rust project.
+  The implementation has to be mature, widely used and must follow all policies
+  on alternate implementation set by the Rust project (if any are set).
+
+Note that when evaluating the eligibility criteria, the spirit of the rules
+will be considered: if the WG thinks an organization that would not be eligible
+otherwise is trying to find a loophole to be able to apply, the WG will have
+the authority to reject the application.
+
```

Comment:
I would have a requirement that an organisation be trustworthy. I realise the WG has final say and that this is impossible to judge objectively, but I think it is useful to have as a guide and to build trust with the community in general that the policy *will* be applied sensibly.

Reply:
Hmm, I'm not opposed to it, but do you have some criteria in mind to decide whether an organization will be trustworthy?

Reply:
I would not, I would leave that totally to the WG's discretion. I would just add that to the text so there is some expectation that this is an aspect the WG would consider when deciding on whether to include a company in the process

Reply:
I'm kinda skeptical of a vague criteria, as it risks becoming the catch-all for the WG applying this policy unevenly. Do you think something like "the organization must not have a history of confidentiality violations" would still cover your concerns?

Reply:
Well, vague is kinda bad, but we shouldn't be afraid of explicitly leaving stuff to people's discretion or require over-precise definitions. We are dealing with people, not code, and people are vague and imprecise. I think having a guideline which sets expectations and helps guide the WG to make policy decisions is a good thing. "the organization must not have a history of confidentiality violations" seems necessary but not sufficient. Like if a cyber-crime group applied for the disclosure, you'd want to say 'no' even if they were really good at keeping secrets.

Reply:
> Like if a cyber-crime group applied for the disclosure, you'd want to say 'no' even if they were really good at keeping secrets.

Ok that's a good example, and it lets me visualize better what you were worried about! I'll have a think about it.